src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemQueryTagService.cs

```diff
@@ -5,23 +5,28 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
 using Microsoft.Health.Dicom.Features.Common;
 
 namespace Microsoft.Health.Dicom.Core.Features.Workitem
 {
     public class WorkitemQueryTagService : IWorkitemQueryTagService, IDisposable
     {
         private readonly IIndexWorkitemStore _indexWorkitemStore;
-        private readonly AsyncCache<IReadOnlyCollection<WorkitemQueryTagStoreEntry>> _queryTagCache;
+        private readonly AsyncCache<IReadOnlyCollection<QueryTag>> _queryTagCache;
```

Comment:
What is AsyncCache?  Why not use MemoryCache?

Reply:
`AsyncCache<T>` is a type I wrote for resolving schema versions. It's meant to be like an `async` version of `Lazy<T>`